### PR TITLE
SSE4_2 has become the minimal requirements as with DPDK

### PR DESCRIPTION
**Pull request by shinae-woo**
SSE4.2 has become the minimal requirements for DPDK. So as to for BESS.

### CODE REVIEW

File: core/utils/simd.h [hunk 1]
```diff
@@ -41,8 +41,8 @@
 #define __ymm_aligned __attribute__((aligned(32)))
 #define __zmm_aligned __attribute__((aligned(64)))
 
-#if !__SSSE3__
-#error CPU must be at least Core 2 or equivalent (SSSE3 required)
+#if !__SSE4_2__
+#error CPU must be at least corei5 or equivalent (SSE4.2 required)
```
Comment (sangjinhan) [inline core/utils/simd.h]: I think `Intel Nehalem` is a more accurate description than `corei5` (i.e., all Core i3 processors also support SSE4_2)

File: core/utils/http_parser.cc [hunk 2]
```diff
@@ -69,29 +67,29 @@
...
-          *ret = -1;                                                          \
-          return NULL;                                                        \
-        }                                                                     \
-      }                                                                       \
-      ++buf;                                                                  \
-      CHECK_EOF();                                                            \
-    }                                                                         \
-    tok = tok_start;                                                          \
-    toklen = buf - tok_start;                                                 \
+#define ADVANCE_TOKEN(tok, toklen)                                       \
```
Comment (sangjinhan) [inline core/utils/http_parser.cc]: Perhaps it's better to minimize changes (except bug patches) of this file, which was copied from some external open-source projects? Deviation from the original code may make code maintenance later.

**Comment by sangjinhan** [comment]
Note: the SSE4.2 requirement is not yet strictly required for the current version of BESS, but we are planning to update to DPDK 17.11 soon.

File: core/utils/http_parser.cc (hunk 2 above)
Comment (shinae-woo) [reply core/utils/http_parser.cc]: I see. I will roll-back the changes for this file.

File: core/utils/simd.h (hunk 1 above)
Comment (shinae-woo) [reply core/utils/simd.h]: Oh, right.

**Comment by sangjinhan** [comment]
Thanks!
tumblr_m94zyfvd6f1qzbvzho1_500

**Pull request merged**